pkgs/development/compilers/open-watcom/default.nix

```diff
@@ -0,0 +1,147 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+
+, withDocs ? false
+, dosbox
+, ghostscript
+, withGUI ? false
+}:
+
+stdenv.mkDerivation rec {
+  pname = "open-watcom-v2";
+  version = "unstable-2021-11-14";
+  name = "${pname}-unwrapped-${version}";
+
+  src = fetchFromGitHub {
+    owner = "open-watcom";
+    repo = "open-watcom-v2";
+    rev = "57ba42ef9e97d074577a3408bb8f28a586c902f7";
+    sha256 = "1p0k7bg33l6fck0j9d5zgdnqa77jhj8w6hx7ba6zyja3ci8k978f";
+  };
+
+  postPatch = ''
+    patchShebangs *.sh
+
+    for dateSource in cmnvars.sh bld/wipfc/configure; do
+      substituteInPlace $dateSource \
+        --replace '`date ' '`date -ud "@$SOURCE_DATE_EPOCH" '
+    done
+    substituteInPlace build/makeinit \
+      --replace '%__CYEAR__' '%OWCYEAR'
+  '' + lib.optionalString (!stdenv.hostPlatform.isDarwin) ''
+    substituteInPlace build/mif/local.mif \
+      --replace '-static' ""
+  '';
+
+  nativeBuildInputs = [ ]
+    ++ lib.optional (withDocs || withGUI) dosbox
+    ++ lib.optional withDocs ghostscript;
+
+  configurePhase = ''
+    runHook preConfigure
+
+    export OWROOT=$(realpath $PWD)
+    export OWTOOLS=${if stdenv.cc.isClang then "CLANG" else "GCC"}
+    export OWDOCBUILD=${if withDocs then "1" else "0"}
+    export OWGHOSTSCRIPTPATH=${lib.optionalString withDocs "${ghostscript}/bin"}
+    export OWGUINOBUILD=${if withGUI then "0" else "1"}
+    export OWNOBUILD=
+    export OWDISTRBUILD=0
+    export OWDOSBOX=${lib.optionalString (withDocs || withGUI) "${dosbox}/bin/dosbox"}
+    export OWVERBOSE=0
+    export OWRELROOT=$out
+
+    source cmnvars.sh
+
+    runHook postConfigure
+  '';
+
+  buildPhase = ''
+    runHook preBuild
+
+    ./build.sh build
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    ./build.sh cprel
+
+    runHook postInstall
+  '';
+
+  # passthru.tests = {
+  #   simple = runCommand "${pname}-test-native" {} ''
+  #     export WATCOM=${open-watcom-v2}
+  #     export PATH=$[lib.strings.concatMapStringsSep ":" (dir: "$WATCOM/$[dir]") bindirs]:$PATH
+  #     export INCLUDE=$WATCOM/${lib.optionalString (!stdenv.hostPlatform.isWindows) "l"}h
```

Comment:
This test worked, but it had to emulate what the wrapper is supposed to be doing envvar-wise. Is it possible to move these kinds of tests into the wrapper instead so they can be reused when I eventually migrate `open-watcom-bin` to it / replace `open-watcom-bin` with a source-built `open-watcom-1_9`? Since using these tests requires me to add the derivation into its own inputs I was struggling with understanding how that might be done. Or should I continue to emulate the wrapper here?

Reply:
Got tests figured out.